### DISCUSSION TITLE
Hello, I am very stupid in this regard. Tried several times, but failed

### DISCUSSION THREAD

**Discussion by cyl2009** [Dineshkarthik/telegram_media_downloader#83] [Q&A] [answered]
Hello, I am very stupid in this regard. I have tried several times, but no success, I would like to ask you,
Is it necessary
vim media_downloader.py,
Change, these few items

api_hash: your_api_hash
api_id: your_api_id
chat_id: telegram_chat_id
last_read_message_id: 0

thank you very much

**Comment by Dineshkarthik** [comment, marked as answer]
can you explain what is the failure you are facing?

**Reply by cyl2009** [reply]
Hello, thank you for your reply. Because I'm too stupid, so please guide me

The server I use, the system is Debian, I installed it step by step like this,

apt-get update

apt-get install git

git clone https://github.com/Dineshkarthik/telegram_media_downloader.git

cd telegram_media_downloader

sudo apt-get install make

apt-get install python3-pip

pip3 install -r requirements.txt

These, I installed them in your own way. I can't install them. I will check some information from the Internet, and then I will continue. Have been installed here

api_hash: your_api_hash
api_id: your_api_id
chat_id: telegram_chat_id
last_read_message_id: 0

I don’t know where to change these, so I used this command


vim media_downloader.py

I changed these lines in this file, because I don’t know how to copy the code, I had to type it out,

last_read_message_id: int = config["0"]        I changed the last_read_message_id here to "0".

I changed these here,


client = pyrogram. Client(
    "media_downlcader",
    api_id=config["My ID"]                        This is api_id
    api_hasth=confige["d3d522efba....."]   This is api_hash

By the way, I also changed it, this

messages_iter = client.iter_history(
    config["njdlingyu"] This is ---chat_id---https://web.telegram.org/#/im?p=@njdlingyu

Enter, find it, and change it. At the end of this file, after saving and exiting, it doesn't seem to be executed. I think I have failed.
This is my whole result, please give me more guidance, thank you. thank you very much.

This is what appears after execution

root@instance-13:~/telegram_media_downloader# python3 media_downloader.py
Telegram Media Downloader v0.3.1, Copyright (C) 2019 Dineshkarthik <https://github.com/Dineshkarthik>
Licensed under the terms of the MIT License

INFO:media_downloader:Device: CPython 3.7.3-Telegram Media Downloader 0.3.1
INFO:media_downloader:System: Linux 4.19.0-14-cloud-amd64 (EN)
Traceback (most recent call last):
  File "media_downloader.py", line 362, in <module>
    main()
  File "media_downloader.py", line 347, in main
    begin_import(config, pagination_limit=100)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "media_downloader.py", line 298, in begin_import
    api_id=config["My ID"],
KeyError:'My ID'

**Reply by Dineshkarthik** [reply]
Hi don't change anything in `media_downloader.py`. 

The following is present in **config.ym**l file.
```api_hash: your_api_hash
api_id: your_api_id
chat_id: telegram_chat_id
last_read_message_id: 0
```

**Reply by cyl2009** [reply]
Thank you very much for your guidance, I have successfully downloaded it,
I would like to ask, after I download the content of the first channel, I want to download the content of another channel, then the content of the two channels will be downloaded together, or will a new folder be created. Thank you again for your help.

**Reply by Dineshkarthik** [reply]
yeah for now they will be downloaded inside the same folder
